### PR TITLE
chore: Excluding certain event_types from processing uid

### PR DESCRIPTION
**Pull request by pragatimodi**
(internal change)

Payload body:
```
{
  "iss": "",
  "aud": "",
  "iat": "",
  "exp": "",
  "event_id": "",
  "event_type": "beforeCreate",
  "user_agent": "",
  "ip_address": "",
  "locale": "",
  "sign_in_method": "",
  "sub": "",
  "user_record": {
    "uid": "NzZjFR0TDsWwxYUuPvkfUfWkZx73",
    "email": "abcd@example.com",
    "email_verified": false,
    "provider_data": [
      {
        "provider_id": "",
        "email": "abcd@example.com",
        "uid": "abcd@example.com"
      }
    ],
    "metadata": {
      "last_sign_in_time": "",
      "creation_time": ""
    }
  }
}
```

Since the event types `beforeSendSms` and `beforeSendEmail` do not include user record in the payload, we exclude them from processing sub which is the same as uid which is a part of user record.

### CODE REVIEW

File: src/auth/token-verifier.ts [hunk 1]
```diff
@@ -529,13 +529,17 @@ export class FirebaseTokenVerifier {
...
         `"${this.issuer}` + projectId + '" but got "' +
         payload.iss + '".' + projectIdMatchMessage + verifyJwtTokenDocsMessage;
-    } else if (typeof payload.sub !== 'string') {
-      errorMessage = `${this.tokenInfo.jwtName} has no "sub" (subject) claim.` + verifyJwtTokenDocsMessage;
-    } else if (payload.sub === '') {
-      errorMessage = `${this.tokenInfo.jwtName} has an empty string "sub" (subject) claim.` + verifyJwtTokenDocsMessage;
-    } else if (payload.sub.length > 128) {
-      errorMessage = `${this.tokenInfo.jwtName} has "sub" (subject) claim longer than 128 characters.` +
-        verifyJwtTokenDocsMessage;
+    } else if (!(payload.event_type !== undefined &&
```
Comment (Xiaoshouzi-gh) [inline src/auth/token-verifier.ts:532]: let's add a comment below this line explain what this is and why we needed it.
You can re-phrase below for reasons - for beforeEmailSend and beforeSMSSend events, there is no user record available.  sub is the same as uid which is part of the user record.

**Review by Xiaoshouzi-gh — APPROVED** [review]
LTGM.
Please hold off on the merging before code-freeze lift.

**Review by Xiaoshouzi-gh — CHANGES_REQUESTED** [review]
are you able to add unit tests for this?

**Review by Xiaoshouzi-gh — APPROVED** [review]
Awesome LGTM

**Review by lahirumaramba — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**